packages/swapper/src/swappers/ChainflipSwapper/endpoints.ts

```diff
@@ -0,0 +1,188 @@
+        // `&minimumPrice=` +
+        // `&refundAddress=${from}` +
+        // `&retryDurationInBlocks=10` +
+        `&commissionBps=${serviceCommission}`,
+
+      // TODO: Below is the reference code of Chainflip to calculate the minPrice parameter
+      //     const tolerance = new BigNumber(params.slippageTolerancePercent);
+      //     const estimatedPrice = new BigNumber(quote.estimatedPrice);
+      //     minPrice = estimatedPrice
+      //       .times(new BigNumber(100).minus(tolerance).dividedBy(100))
+      //       .toFixed(assetConstants[destAsset].decimals);
+
+      // TODO: For DCA swaps we need to add the numberOfChunks/chunkIntervalBlocks parameters
+    )
+
+    if (maybeSwapResponse.isErr()) {
+      const error = maybeSwapResponse.unwrapErr()
+      const cause = error.cause as AxiosError<any, any>
+      throw Error(cause.response!.data.detail)
+    }
+
+    const { data: swapResponse } = maybeSwapResponse.unwrap()
+
+    if (!swapResponse.id) throw Error('missing swap ID')
+
+    tradeQuoteMetadata.set(tradeQuote.id, swapResponse)
+
+    const depositAddress = swapResponse.address!
+    const { assetReference } = fromAssetId(step.sellAsset.assetId)
+    const isTokenSend = isToken(step.sellAsset.assetId)
+
+    const adapter = assertGetEvmChainAdapter(step.sellAsset.chainId)
+
+    const getFeeDataInput: GetFeeDataInput<EvmChainId> = {
+      to: depositAddress,
+      value: step.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      chainSpecific: {
+        from,
+        contractAddress: isTokenSend ? assetReference : undefined,
+        data: undefined,
+      },
+      sendMax: false,
+    }
+    const feeData = await adapter.getFeeData(getFeeDataInput)
+    const fees = feeData[FeeDataKey.Average]
+
+    if (!isExecutableTradeStep(step)) throw Error('Unable to execute trade step')
+
+    const unsignedTxInput = await adapter.buildSendApiTransaction({
+      to: depositAddress,
+      from,
+      value: step.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      accountNumber: step.accountNumber,
+      chainSpecific: {
+        gasLimit: fees.chainSpecific.gasLimit,
+        contractAddress: isTokenSend ? assetReference : undefined,
+        ...(supportsEIP1559
+          ? {
+              maxFeePerGas: fees.chainSpecific.maxFeePerGas!,
+              maxPriorityFeePerGas: fees.chainSpecific.maxPriorityFeePerGas!,
+            }
+          : {
+              gasPrice: fees.chainSpecific.gasPrice,
+            }),
+      },
+    })
+
+    return {
+      chainId: Number(fromChainId(chainId).chainReference),
+      data: unsignedTxInput.data,
+      to: unsignedTxInput.to,
+      from,
+      value: unsignedTxInput.value,
+      gasLimit: unsignedTxInput.gasLimit,
+      maxFeePerGas: unsignedTxInput.maxFeePerGas,
+      maxPriorityFeePerGas: unsignedTxInput.maxPriorityFeePerGas,
+      gasPrice: unsignedTxInput.gasPrice,
+    }
+  },
+  getUnsignedUtxoTransaction,
```

Comment:
This should come back to this module as well since we'll need to add to trade quote meta for status polling